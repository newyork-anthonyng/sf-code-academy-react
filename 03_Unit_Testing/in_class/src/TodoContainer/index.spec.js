import React from "react";
import Enzyme, { shallow, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import toJSON from "enzyme-to-json";
import TodoItemContainer from "./";

Enzyme.configure({ adapter: new Adapter() });

// Add new tests here
it("should return true", function() {
  expect(true).toEqual(true);
});

it("should toggle the completed state of a TodoItem when I click on the TodoItem", () => {
  const wrapper = mount(<TodoItemContainer />);

  const listElement = wrapper.find("li").at(0);
  listElement.simulate("click");

  expect(toJSON(wrapper)).toMatchSnapshot();
});

it("should input element when I type in it", () => {
  const wrapper = shallow(<TodoItemContainer />);

  const inputElement = wrapper.find("input");
  inputElement.simulate("change", { target: { value: "Expedia" }});

  expect(toJSON(wrapper)).toMatchSnapshot();
});

it("should add a new todo item when I click on the button", () => {
  const wrapper = shallow(<TodoItemContainer />);

  const inputElement = wrapper.find("input");
  inputElement.simulate("change", { target: { value: "Expedia" }});

  const buttonElement = wrapper.find("button");
  buttonElement.simulate("click");

  expect(toJSON(wrapper)).toMatchSnapshot();
});
