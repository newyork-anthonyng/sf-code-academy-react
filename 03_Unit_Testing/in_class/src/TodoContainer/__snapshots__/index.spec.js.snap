// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should add a new todo item when I click on the button 1`] = `
<div>
  <div>
    <input
      onChange={[Function]}
      type="text"
      value="Expedia"
    />
    <button
      onClick={[Function]}
      type="button"
    >
      Add todo item
    </button>
  </div>
  <TodoList
    onTodoItemClick={[Function]}
    todoItems={
      Array [
        Object {
          "completed": true,
          "text": "Walk dog",
        },
        Object {
          "completed": true,
          "text": "Learn react",
        },
        Object {
          "completed": false,
          "text": "Expedia",
        },
      ]
    }
  />
</div>
`;

exports[`should input element when I type in it 1`] = `
<div>
  <div>
    <input
      onChange={[Function]}
      type="text"
      value="Expedia"
    />
    <button
      onClick={[Function]}
      type="button"
    >
      Add todo item
    </button>
  </div>
  <TodoList
    onTodoItemClick={[Function]}
    todoItems={
      Array [
        Object {
          "completed": true,
          "text": "Walk dog",
        },
        Object {
          "completed": true,
          "text": "Learn react",
        },
      ]
    }
  />
</div>
`;

exports[`should toggle the completed state of a TodoItem when I click on the TodoItem 1`] = `
<TodoContainer>
  <div>
    <div>
      <input
        onChange={[Function]}
        type="text"
        value=""
      />
      <button
        onClick={[Function]}
        type="button"
      >
        Add todo item
      </button>
    </div>
    <TodoList
      onTodoItemClick={[Function]}
      todoItems={
        Array [
          Object {
            "completed": false,
            "text": "Walk dog",
          },
          Object {
            "completed": true,
            "text": "Learn react",
          },
        ]
      }
    >
      <ul>
        <TodoItem
          completed={false}
          index={0}
          key="0"
          onClick={[Function]}
          text="Walk dog"
        >
          <li
            onClick={[Function]}
            style={
              Object {
                "color": "black",
                "cursor": "pointer",
                "textDecoration": "",
              }
            }
          >
            Walk dog
          </li>
        </TodoItem>
        <TodoItem
          completed={true}
          index={1}
          key="1"
          onClick={[Function]}
          text="Learn react"
        >
          <li
            onClick={[Function]}
            style={
              Object {
                "color": "green",
                "cursor": "pointer",
                "textDecoration": "line-through",
              }
            }
          >
            Learn react
          </li>
        </TodoItem>
      </ul>
    </TodoList>
  </div>
</TodoContainer>
`;
